bin/metadata_validation.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+
+import json
+
+import synapseclient
+from synapseclient import Entity
+
+
+def get_synaspe_file(syn_id):
+    """
+    Initializes synapseclient session and retrieves file metadata
+    """
+    syn = synapseclient.Synapse()
+    syn.login(authToken=os.environ["SYNAPSE_AUTH_TOKEN"])
+    syn_file = syn.get(syn_id)
+    return syn_file
+
+def json_dump(syn_id, test, data):
+    """
+    Dumps dictionary into json file
+    """
+    with open(r'{0}-{1}.json'.format(syn_id, test), "w") as outfile:
+        json.dump(data, outfile)
+
+def execute_file_entity_test(syn_id, syn_file, md5_checksum):
+    """
+    1. Creates file_entity_test dictionary for eventual json dump
+    2. Checks synapse file for concreteType if FileEntity
+    3. Calls json_dump to create json file with validation result and input metadata
+    """
+    file_entity_test = {
+        "input": {
+            "synapse_id": syn_id,
+            "type": syn_file.get("concreteType"),
+            "version_number": syn_file.properties.get("versionNumber"),
+            "md5_checksum": md5_checksum,
+        },
+        "test": {
+            "name": "file_entity_test", 
+            "status": "",
+            "reason": None}
+    }
+
+    if syn_file.get("concreteType") == "org.sagebionetworks.repo.model.FileEntity":
+        file_entity_test["test"]["status"] = "pass"
+    else:
+        file_entity_test["test"]["status"] = "fail"
+
+    json_dump(syn_id, "file_entity_test", file_entity_test)
```

Comment:
**Nit (_i.e._ [optional](https://google.github.io/eng-practices/review/reviewer/standard.html#:~:text=Reviewers%20should%20always%20feel%20free,they%20could%20choose%20to%20ignore.)):** By performing the test first, you can avoid the nested dictionary assignments later. I'm also suggesting that you store the past part of the concrete type instead of the full name, which is an artifact of Java. You can check [here](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/c306a4ae924e3b017860009f63c19a8a8d8d2f1f/synapseclient/core/constants/concrete_types.py) that the last part is sufficient for identifying the entity type. 

```suggestion
    # Perform the file entity test
    concrete_type = syn_file.get("concreteType", "")
    entity_type = concrete_type.split(".")[-1]  # Last part of concrete type
    file_entity_test = entity_type == "FileEntity"
    file_entity_status = "pass" if file_entity_test else "fail"

    # Prepare the test data for output
    file_entity_data = {
        "input": {
            "synapse_id": syn_id,
            "type": entity_type,
            "version_number": syn_file.properties.get("versionNumber"),
            "md5_checksum": md5_checksum,
        },
        "test": {
            "name": "file_entity_test", 
            "status": file_entity_status,
            "reason": None
        }
    }

    # Output the test data
    json_dump(syn_id, "file_entity_test", file_entity_data)
```